models/balancer/arbitrum/balancer_v2_arbitrum_bpt_supply.sql

```diff
@@ -0,0 +1,523 @@
+                            THEN 0x00 
+                            ELSE pool_token 
+                        END
+                    ELSE pool_token
+                END
+            ] AS tokens_in
+            , ARRAY[
+                CASE
+                    WHEN delta_pool_token <= INT256 '0' 
+                    THEN
+                        CASE 
+                            WHEN lp_token_burned > INT256 '0' 
+                            THEN 0x00 
+                            ELSE pool_token 
+                        END
+                    ELSE pool_token
+                END
+            ] AS tokens_out
+            , ARRAY[
+                CASE 
+                    WHEN delta_pool_token > INT256 '0' 
+                    THEN lp_token_minted
+                    ELSE lp_token_minted
+                END
+            ] AS amounts_in
+            , ARRAY[
+                CASE 
+                    WHEN delta_pool_token > INT256 '0' 
+                    THEN delta_pool_token + lp_token_burned
+                    ELSE lp_token_burned
+                END
+            ] AS amounts_out
+        FROM ( --Start subquery
+            SELECT 
+                x.contract_address
+                , x.evt_tx_hash
+                , x.evt_index
+                , x.evt_block_time
+                , x.evt_block_number
+                , x.poolId AS poolId
+                , sp.pool_token AS pool_token
+                , x.deltas
+                , x.deltas[ARRAY_POSITION(x.tokens, sp.pool_token)] AS delta_pool_token
+                , y.lp_token_burned
+                , y.lp_token_minted
+            FROM {{ source('balancer_v2_arbitrum', 'Vault_evt_PoolBalanceChanged') }} x --#########################
+            INNER JOIN (SELECT poolId, tokens, pool_token FROM sub_pools) sp
+                ON sp.poolId = x.poolId
+            INNER JOIN (
+                SELECT 
+                    evt_tx_hash
+                    , contract_address
+                    , IF(
+                        sum(CAST(value AS int256)) FILTER(
+                            WHERE "from" = 0x0000000000000000000000000000000000000000
+                        ) IS NULL
+                        , int256 '0'
+                        , sum(CAST(value AS int256)) FILTER(
+                            WHERE "from" = 0x0000000000000000000000000000000000000000
+                        )
+                    ) AS lp_token_minted
+                    , IF(
+                        sum(CAST(value AS int256)) FILTER(
+                            WHERE to = 0x0000000000000000000000000000000000000000
+                        ) IS NULL
+                        , int256 '0'
+                        , sum(CAST(value AS int256)) FILTER(
+                            WHERE to = 0x0000000000000000000000000000000000000000
+                        )
+                    ) AS lp_token_burned
+                FROM {{ ref('evms_erc20_transfers') }}  --#########################
```

Comment:
at this time, we are not accepting usage of the `evms` schema views in downstream spells. we're aware that these views are popular to query on the dune app, but we aren't materializing due to resource constraints on the engine for regular spellbook orchestration updates. we have raised this to the team internally of the interest in having these cross-chain evm views materialized, but not sure the plan quite yet.

with that said, since your spells are blockchain-specific anyway, can you change them all to use the blockchain-specific source table instead?

Reply:
this would apply to all within that schema, except potentially `evms.info`, as that is a metadata driven spell.

Reply:
No problem, I have made the requested changes.